// wp-types/index.js
/* eslint-disable */ /**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */ export var WP_Comment_Type_Name = /*#__PURE__*/ function(WP_Comment_Type_Name) {
    WP_Comment_Type_Name["comment"] = "comment";
    WP_Comment_Type_Name["pingback"] = "pingback";
    WP_Comment_Type_Name["trackback"] = "trackback";
    return WP_Comment_Type_Name;
}({});
export var WP_Post_Type_Name = /*#__PURE__*/ function(WP_Post_Type_Name) {
    WP_Post_Type_Name["post"] = "post";
    WP_Post_Type_Name["page"] = "page";
    WP_Post_Type_Name["attachment"] = "attachment";
    WP_Post_Type_Name["revision"] = "revision";
    WP_Post_Type_Name["nav_menu_item"] = "nav_menu_item";
    WP_Post_Type_Name["custom_css"] = "custom_css";
    WP_Post_Type_Name["customize_changeset"] = "customize_changeset";
    WP_Post_Type_Name["oembed_cache"] = "oembed_cache";
    WP_Post_Type_Name["user_request"] = "user_request";
    WP_Post_Type_Name["wp_block"] = "wp_block";
    WP_Post_Type_Name["wp_global_styles"] = "wp_global_styles";
    WP_Post_Type_Name["wp_navigation"] = "wp_navigation";
    WP_Post_Type_Name["wp_template"] = "wp_template";
    WP_Post_Type_Name["wp_template_part"] = "wp_template_part";
    return WP_Post_Type_Name;
}({});
export var WP_Taxonomy_Name = /*#__PURE__*/ function(WP_Taxonomy_Name) {
    WP_Taxonomy_Name["category"] = "category";
    WP_Taxonomy_Name["post_tag"] = "post_tag";
    WP_Taxonomy_Name["nav_menu"] = "nav_menu";
    WP_Taxonomy_Name["post_format"] = "post_format";
    WP_Taxonomy_Name["wp_template_part_area"] = "wp_template_part_area";
    WP_Taxonomy_Name["wp_theme"] = "wp_theme";
    return WP_Taxonomy_Name;
}({});
export var WP_Post_Status_Name = /*#__PURE__*/ function(WP_Post_Status_Name) {
    WP_Post_Status_Name["publish"] = "publish";
    WP_Post_Status_Name["draft"] = "draft";
    WP_Post_Status_Name["auto_draft"] = "auto-draft";
    WP_Post_Status_Name["inherit"] = "inherit";
    WP_Post_Status_Name["pending"] = "pending";
    WP_Post_Status_Name["future"] = "future";
    WP_Post_Status_Name["trash"] = "trash";
    WP_Post_Status_Name["private"] = "private";
    return WP_Post_Status_Name;
}({});
export var WP_Post_Comment_Status_Name = /*#__PURE__*/ function(WP_Post_Comment_Status_Name) {
    WP_Post_Comment_Status_Name["open"] = "open";
    WP_Post_Comment_Status_Name["closed"] = "closed";
    return WP_Post_Comment_Status_Name;
}({});
export var WP_Object_Filter_Context = /*#__PURE__*/ function(WP_Object_Filter_Context) {
    WP_Object_Filter_Context["attribute"] = "attribute";
    WP_Object_Filter_Context["db"] = "db";
    WP_Object_Filter_Context["display"] = "display";
    WP_Object_Filter_Context["edit"] = "edit";
    WP_Object_Filter_Context["js"] = "js";
    WP_Object_Filter_Context["raw"] = "raw";
    WP_Object_Filter_Context["rss"] = "rss";
    return WP_Object_Filter_Context;
}({});
export var WP_User_Role_Name = /*#__PURE__*/ function(WP_User_Role_Name) {
    WP_User_Role_Name["administrator"] = "administrator";
    WP_User_Role_Name["editor"] = "editor";
    WP_User_Role_Name["author"] = "author";
    WP_User_Role_Name["contributor"] = "contributor";
    WP_User_Role_Name["subscriber"] = "subscriber";
    return WP_User_Role_Name;
}({});
export var WP_Comment_Status_Name = /*#__PURE__*/ function(WP_Comment_Status_Name) {
    WP_Comment_Status_Name["approved"] = "approved";
    WP_Comment_Status_Name["unapproved"] = "unapproved";
    WP_Comment_Status_Name["spam"] = "spam";
    WP_Comment_Status_Name["trash"] = "trash";
    return WP_Comment_Status_Name;
}({});
export var WP_Post_Format_Name = /*#__PURE__*/ function(WP_Post_Format_Name) {
    WP_Post_Format_Name["aside"] = "aside";
    WP_Post_Format_Name["audio"] = "audio";
    WP_Post_Format_Name["chat"] = "chat";
    WP_Post_Format_Name["gallery"] = "gallery";
    WP_Post_Format_Name["image"] = "image";
    WP_Post_Format_Name["link"] = "link";
    WP_Post_Format_Name["quote"] = "quote";
    WP_Post_Format_Name["standard"] = "standard";
    WP_Post_Format_Name["status"] = "status";
    WP_Post_Format_Name["video"] = "video";
    return WP_Post_Format_Name;
}({});
export var WP_Http_Status_Code = /*#__PURE__*/ function(WP_Http_Status_Code) {
    WP_Http_Status_Code[WP_Http_Status_Code["HTTP_CONTINUE"] = 100] = "HTTP_CONTINUE";
    WP_Http_Status_Code[WP_Http_Status_Code["SWITCHING_PROTOCOLS"] = 101] = "SWITCHING_PROTOCOLS";
    WP_Http_Status_Code[WP_Http_Status_Code["PROCESSING"] = 102] = "PROCESSING";
    WP_Http_Status_Code[WP_Http_Status_Code["EARLY_HINTS"] = 103] = "EARLY_HINTS";
    WP_Http_Status_Code[WP_Http_Status_Code["OK"] = 200] = "OK";
    WP_Http_Status_Code[WP_Http_Status_Code["CREATED"] = 201] = "CREATED";
    WP_Http_Status_Code[WP_Http_Status_Code["ACCEPTED"] = 202] = "ACCEPTED";
    WP_Http_Status_Code[WP_Http_Status_Code["NON_AUTHORITATIVE_INFORMATION"] = 203] = "NON_AUTHORITATIVE_INFORMATION";
    WP_Http_Status_Code[WP_Http_Status_Code["NO_CONTENT"] = 204] = "NO_CONTENT";
    WP_Http_Status_Code[WP_Http_Status_Code["RESET_CONTENT"] = 205] = "RESET_CONTENT";
    WP_Http_Status_Code[WP_Http_Status_Code["PARTIAL_CONTENT"] = 206] = "PARTIAL_CONTENT";
    WP_Http_Status_Code[WP_Http_Status_Code["MULTI_STATUS"] = 207] = "MULTI_STATUS";
    WP_Http_Status_Code[WP_Http_Status_Code["IM_USED"] = 226] = "IM_USED";
    WP_Http_Status_Code[WP_Http_Status_Code["MULTIPLE_CHOICES"] = 300] = "MULTIPLE_CHOICES";
    WP_Http_Status_Code[WP_Http_Status_Code["MOVED_PERMANENTLY"] = 301] = "MOVED_PERMANENTLY";
    WP_Http_Status_Code[WP_Http_Status_Code["FOUND"] = 302] = "FOUND";
    WP_Http_Status_Code[WP_Http_Status_Code["SEE_OTHER"] = 303] = "SEE_OTHER";
    WP_Http_Status_Code[WP_Http_Status_Code["NOT_MODIFIED"] = 304] = "NOT_MODIFIED";
    WP_Http_Status_Code[WP_Http_Status_Code["USE_PROXY"] = 305] = "USE_PROXY";
    WP_Http_Status_Code[WP_Http_Status_Code["RESERVED"] = 306] = "RESERVED";
    WP_Http_Status_Code[WP_Http_Status_Code["TEMPORARY_REDIRECT"] = 307] = "TEMPORARY_REDIRECT";
    WP_Http_Status_Code[WP_Http_Status_Code["PERMANENT_REDIRECT"] = 308] = "PERMANENT_REDIRECT";
    WP_Http_Status_Code[WP_Http_Status_Code["BAD_REQUEST"] = 400] = "BAD_REQUEST";
    WP_Http_Status_Code[WP_Http_Status_Code["UNAUTHORIZED"] = 401] = "UNAUTHORIZED";
    WP_Http_Status_Code[WP_Http_Status_Code["PAYMENT_REQUIRED"] = 402] = "PAYMENT_REQUIRED";
    WP_Http_Status_Code[WP_Http_Status_Code["FORBIDDEN"] = 403] = "FORBIDDEN";
    WP_Http_Status_Code[WP_Http_Status_Code["NOT_FOUND"] = 404] = "NOT_FOUND";
    WP_Http_Status_Code[WP_Http_Status_Code["METHOD_NOT_ALLOWED"] = 405] = "METHOD_NOT_ALLOWED";
    WP_Http_Status_Code[WP_Http_Status_Code["NOT_ACCEPTABLE"] = 406] = "NOT_ACCEPTABLE";
    WP_Http_Status_Code[WP_Http_Status_Code["PROXY_AUTHENTICATION_REQUIRED"] = 407] = "PROXY_AUTHENTICATION_REQUIRED";
    WP_Http_Status_Code[WP_Http_Status_Code["REQUEST_TIMEOUT"] = 408] = "REQUEST_TIMEOUT";
    WP_Http_Status_Code[WP_Http_Status_Code["CONFLICT"] = 409] = "CONFLICT";
    WP_Http_Status_Code[WP_Http_Status_Code["GONE"] = 410] = "GONE";
    WP_Http_Status_Code[WP_Http_Status_Code["LENGTH_REQUIRED"] = 411] = "LENGTH_REQUIRED";
    WP_Http_Status_Code[WP_Http_Status_Code["PRECONDITION_FAILED"] = 412] = "PRECONDITION_FAILED";
    WP_Http_Status_Code[WP_Http_Status_Code["REQUEST_ENTITY_TOO_LARGE"] = 413] = "REQUEST_ENTITY_TOO_LARGE";
    WP_Http_Status_Code[WP_Http_Status_Code["REQUEST_URI_TOO_LONG"] = 414] = "REQUEST_URI_TOO_LONG";
    WP_Http_Status_Code[WP_Http_Status_Code["UNSUPPORTED_MEDIA_TYPE"] = 415] = "UNSUPPORTED_MEDIA_TYPE";
    WP_Http_Status_Code[WP_Http_Status_Code["REQUESTED_RANGE_NOT_SATISFIABLE"] = 416] = "REQUESTED_RANGE_NOT_SATISFIABLE";
    WP_Http_Status_Code[WP_Http_Status_Code["EXPECTATION_FAILED"] = 417] = "EXPECTATION_FAILED";
    WP_Http_Status_Code[WP_Http_Status_Code["IM_A_TEAPOT"] = 418] = "IM_A_TEAPOT";
    WP_Http_Status_Code[WP_Http_Status_Code["MISDIRECTED_REQUEST"] = 421] = "MISDIRECTED_REQUEST";
    WP_Http_Status_Code[WP_Http_Status_Code["UNPROCESSABLE_ENTITY"] = 422] = "UNPROCESSABLE_ENTITY";
    WP_Http_Status_Code[WP_Http_Status_Code["LOCKED"] = 423] = "LOCKED";
    WP_Http_Status_Code[WP_Http_Status_Code["FAILED_DEPENDENCY"] = 424] = "FAILED_DEPENDENCY";
    WP_Http_Status_Code[WP_Http_Status_Code["TOO_EARLY"] = 425] = "TOO_EARLY";
    WP_Http_Status_Code[WP_Http_Status_Code["UPGRADE_REQUIRED"] = 426] = "UPGRADE_REQUIRED";
    WP_Http_Status_Code[WP_Http_Status_Code["PRECONDITION_REQUIRED"] = 428] = "PRECONDITION_REQUIRED";
    WP_Http_Status_Code[WP_Http_Status_Code["TOO_MANY_REQUESTS"] = 429] = "TOO_MANY_REQUESTS";
    WP_Http_Status_Code[WP_Http_Status_Code["REQUEST_HEADER_FIELDS_TOO_LARGE"] = 431] = "REQUEST_HEADER_FIELDS_TOO_LARGE";
    WP_Http_Status_Code[WP_Http_Status_Code["UNAVAILABLE_FOR_LEGAL_REASONS"] = 451] = "UNAVAILABLE_FOR_LEGAL_REASONS";
    WP_Http_Status_Code[WP_Http_Status_Code["INTERNAL_SERVER_ERROR"] = 500] = "INTERNAL_SERVER_ERROR";
    WP_Http_Status_Code[WP_Http_Status_Code["NOT_IMPLEMENTED"] = 501] = "NOT_IMPLEMENTED";
    WP_Http_Status_Code[WP_Http_Status_Code["BAD_GATEWAY"] = 502] = "BAD_GATEWAY";
    WP_Http_Status_Code[WP_Http_Status_Code["SERVICE_UNAVAILABLE"] = 503] = "SERVICE_UNAVAILABLE";
    WP_Http_Status_Code[WP_Http_Status_Code["GATEWAY_TIMEOUT"] = 504] = "GATEWAY_TIMEOUT";
    WP_Http_Status_Code[WP_Http_Status_Code["HTTP_VERSION_NOT_SUPPORTED"] = 505] = "HTTP_VERSION_NOT_SUPPORTED";
    WP_Http_Status_Code[WP_Http_Status_Code["VARIANT_ALSO_NEGOTIATES"] = 506] = "VARIANT_ALSO_NEGOTIATES";
    WP_Http_Status_Code[WP_Http_Status_Code["INSUFFICIENT_STORAGE"] = 507] = "INSUFFICIENT_STORAGE";
    WP_Http_Status_Code[WP_Http_Status_Code["NOT_EXTENDED"] = 510] = "NOT_EXTENDED";
    WP_Http_Status_Code[WP_Http_Status_Code["NETWORK_AUTHENTICATION_REQUIRED"] = 511] = "NETWORK_AUTHENTICATION_REQUIRED";
    return WP_Http_Status_Code;
}({});
